javagen/src/main/java/com/azure/autorest/mapper/ClientMethodMapper.java

```diff
@@ -284,22 +275,8 @@ private List<ClientMethod> createClientMethods(Operation operation, boolean isPr
                     methodTransformationDetails.add(new MethodTransformationDetail(outParameter, new ArrayList<>()));
                 }
 
-                if (isProtocolMethod) {
-                    ClientMethodParameter requestOptions = new ClientMethodParameter.Builder()
-                        .description("The options to configure the HTTP request before HTTP client sends it")
-                        .wireType(ClassType.RequestOptions)
-                        .name("requestOptions")
-                        .location(RequestParameterLocation.NONE)
-                        .constant(false)
-                        .required(true)
-                        .fromClient(false)
-                        .annotations(Collections.emptyList())
-                        .build();
-                    parameters.add(requestOptions);
-                }
-
-                final boolean generateClientMethodWithOnlyRequiredParameters
-                    = settings.getRequiredParameterClientMethods() && hasNonRequiredParameters(parameters);
+                final MethodOverloadType defaultOverloadType = hasNonRequiredParameters(parameters) ? MethodOverloadType.OVERLOAD_MAXIMUM : MethodOverloadType.OVERLOAD_MINIMUM_MAXIMUM;
+                final boolean generateOnlyRequiredParameters = settings.getRequiredParameterClientMethods() && defaultOverloadType == MethodOverloadType.OVERLOAD_MAXIMUM;
```

Comment:
settings.isRequiredParameterClientMethods()?

Reply:
which variable do you mean?

Reply:
generateOnlyRequiredParameters only happen when setting sets that, AND that overload is needed (which is not if there is no optional parameters in API)

Reply:
OK, I will rename it. Thx